apps/src/code-studio/components/ExportAllowedDialog.jsx

```diff
@@ -0,0 +1,744 @@
+      (this.props.appType === 'applab' || this.props.appType === 'gamelab');
+    return (
+      <div>
+        <BaseDialog
+          style={styles.modal}
+          isOpen={this.props.isOpen}
+          handleClose={this.close}
+          hideBackdrop={this.props.hideBackdrop}
+        >
+          {this.sharingDisabled() && (
+            <div style={{position: 'relative'}}>
+              <div style={{paddingRight: 10}}>
+                <p>{i18n.sharingBlockedByTeacher()}</p>
+              </div>
+              <div style={{clear: 'both', height: 40}}>
+                <button
+                  type="button"
+                  id="continue-button"
+                  style={{position: 'absolute', right: 0, bottom: 0, margin: 0}}
+                  onClick={this.close}
+                >
+                  {i18n.dialogOK()}
+                </button>
+              </div>
+            </div>
+          )}
+          {!this.sharingDisabled() && (
+            <div>
+              <div
+                id="project-export"
+                className="modal-content no-modal-icon"
+                style={{position: 'relative'}}
+              >
+                <p className="dialog-title">Export your project</p>
+                {this.props.isAbusive && (
+                  <AbuseError
+                    i18n={{
+                      tos: this.props.i18n.t('project.abuse.tos'),
+                      contact_us: this.props.i18n.t('project.abuse.contact_us')
+                    }}
+                    className="alert-error"
+                    style={styles.abuseStyle}
+                    textStyle={styles.abuseTextStyle}
+                  />
+                )}
+                {showShareWarning && (
+                  <p style={styles.shareWarning}>
+                    {this.props.i18n.t('project.share_u13_warning')}
+                  </p>
+                )}
+                {this.renderMainContent()}
+                <div style={styles.buttonRow}>
+                  <button
+                    type="button"
+                    style={styles.cancelButton}
+                    onClick={this.close}
+                  >
+                    Cancel
+                  </button>
+                  <button
+                    type="button"
+                    style={
+                      actionEnabled
+                        ? styles.actionButton
+                        : styles.actionButtonDisabled
+                    }
+                    onClick={this.onActionButton}
+                    disabled={!actionEnabled}
+                  >
+                    {actionText}
+                  </button>
+                </div>
+              </div>
+            </div>
+          )}
+        </BaseDialog>
+      </div>
+    );
+  }
+}
+
+export const UnconnectedExportAllowedDialog = ExportAllowedDialog;
+
+export default connect(
+  state => ({
+    allowExportExpo: state.pageConstants.allowExportExpo || false,
```

Comment:
isn't the `|| false` here redundant?

Reply:
It ensures that we satisfy a `PropTypes` requirement